draft-polli-ratelimit-headers.md

```diff
@@ -658,6 +659,66 @@ Ratelimit-Reset: 50
 {"hello": "world"}
 ~~~
 
+### Dynamic limits for pushing back and slowing down
```

Comment:
The remaining units being `0` is actually one of two possible ways to slow down clients. This one is forcing them to stop and wait a little, so it acts as a hard limit at that point (need to wait), whereas the previous example (which this one follows up on) shows a case in which the server lowers an initial window of 60 seconds with 100 units halfway through to considerably reduce the available units to 10 (9 after the request is responded) so that the client knows to use them during the rest of that window, which reduces the average request rate over that period but still accepts consuming all the remaining units right away. In this latter case the server might just decide the spike would be unacceptable and tell the client to wait a little, like the example shown here:

"Wait for 20 seconds without requests, then the next window is going to have only 15 units for 20 seconds". Presumably once the resource starvation is over the server could carry on with the previous limits, returning to a windows of 60 seconds with 100 units.